oaknational/Oak-Web-Application
# [ENG-275] Fix/tier listing not found

## Description

- handle "no results" e[rror from `tierListing` as `not found` rather](https://github.com/oaknational/Oak-Web-Application/pull/1756/files#diff-9102237cf78f2934724069c454b3b0f3437e5e12a5a0feccdedd941051bba34aR470-R475) than `ZodError`
- set all beta getStaticProps to return `paths: []`
  - to tell Netlify in no uncertain terms not to try to generate pages at build time
  - also because we've not been maintaining the getPaths functions in light of the teachers-2023 changes, so best to remove them, otherwise they represent tech debt
- remove all references got `curriculumApi.get...Paths()` functions


## How to test

1. Go to https://deploy-preview-1757--oak-web-application.netlify.thenational.academy/beta/teachers-2023/key-stages/ks4/subjects/NOT_A_SUBJECT/programmes
2. It **should** return 404 but because of Netlify flaws, it will return Internal server error in preview env 



## Checklist
